In [1]:
from utils import load_finetuned_qwen, round_down_to_2_decimals
from keyword_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
def print_model_scores(model):
    scores = get_scores(model)
    print(f"Mean reciprocal rank: {round_down_to_2_decimals(scores["mean_reciprocal_rank"]):.2f}")
    print(f"Recall@15: {round_down_to_2_decimals(scores["recall_at_15"]):.2f}")
    print(f"Precision@15: {round_down_to_2_decimals(scores["precision_at_15"]):.2f}")
    print(f"F1@15: {round_down_to_2_decimals(scores["f1_at_15"]):.2f}")

Base scibert

In [3]:
print_model_scores("allenai/scibert_scivocab_uncased")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 23850.34it/s]
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.seq_relationship.bias                  | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  | 
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 

Notes:
- UNEXPECTED:	can be ignored when loading from different task/architecture; not ok if you expect identical arch.


Mean reciprocal rank: 0.32
Recall@15: 0.80
Precision@15: 0.13
F1@15: 0.23


Vanilla scibert

In [4]:
print_model_scores("models/vanilla-scibert")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3901.86it/s]


Mean reciprocal rank: 0.55
Recall@15: 0.99
Precision@15: 0.53
F1@15: 0.69


Jargon-layman scibert

In [5]:
print_model_scores("models/jargon-layman-scibert-1.00")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3877.86it/s]


Mean reciprocal rank: 0.58
Recall@15: 0.99
Precision@15: 0.63
F1@15: 0.77


Vanilla qwen

In [6]:
print_model_scores(
    load_finetuned_qwen("models/vanilla-qwen")
)

Loading weights:   0%|          | 0/310 [00:00<?, ?it/s]/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/bitsandbytes/backends/cuda/ops.py:213: FutureWarning: _check_is_size will be removed in a future PyTorch release along with guard_size_oblivious.     Use _check(i >= 0) instead.
  torch._check_is_size(blocksize)
Loading weights: 100%|██████████| 310/310 [00:00<00:00, 737.14it/s]


Mean reciprocal rank: 0.58
Recall@15: 0.96
Precision@15: 0.23
F1@15: 0.37


Jargon-layman qwen

In [7]:
print_model_scores(
    load_finetuned_qwen("models/jargon-layman-qwen")
)

Loading weights:   0%|          | 0/310 [00:00<?, ?it/s]/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/bitsandbytes/backends/cuda/ops.py:213: FutureWarning: _check_is_size will be removed in a future PyTorch release along with guard_size_oblivious.     Use _check(i >= 0) instead.
  torch._check_is_size(blocksize)
Loading weights: 100%|██████████| 310/310 [00:00<00:00, 858.31it/s]


Mean reciprocal rank: 0.58
Recall@15: 0.96
Precision@15: 0.23
F1@15: 0.37


# Results

| Model | MRR | Recall@15 | Precision@15 |
|---|---|---|---|
| allenai/scibert_scivocab_uncased | 0.32 | 0.80 | 0.13 |
| vanilla-scibert | <u>0.55</u> | **0.99** | <u>0.53</u> |
| jargon-layman-scibert | **0.58** | **0.99** | **0.63** |
| vanilla-qwen | **0.58** | <u>0.96</u> | 0.23 |
| jargon-layman-qwen | **0.58** | <u>0.96</u> | 0.23 |